site/source/docs/api_reference/emscripten.h.rst

```diff
@@ -931,6 +931,15 @@ Functions
 	:returns: A pointer to the preloaded image or NULL.
 	:rtype: char*
 
+.. c:function:: char *emscripten_print_double(double x, char *to)
+
+	Prints a double as a string. This is useful because JS engines have good support for printing out a double in a way that takes the least possible size, but preserves all the information in the double, i.e., it can then be parsed back in a perfectly reversible manner (snprintf etc. do not do so, sadly).
+
+	No zero terminator is emitted, just the double itself.
```

Comment:
There is potential for a bug if user passes the generated string to some other function that expects a string. Always zero-terminating is safer and more secure, and other number->string conversion functions generally do so as well (e.g. http://www.cplusplus.com/reference/cstdlib/itoa/ ), and avoids the need for the user to remember to patch up later like this: https://github.com/kripken/emscripten/pull/3766/files#diff-5f116af1e8e0aaab0009e915f331c283R20 .